Update mdpow-pow and mdpow-pcw with new input

VOD555 opened the pull request:
Fix #138 #139 
-remove unsurpported input for mdpow-analysis in `mdpow-pow` and `mdpow-pcw`
-set the default plot option to be `none`
-support alchemlyb analysis in `mdpow-pcw`

orbeckst commented on scripts/mdpow-pcw:
```diff
@@ -96,9 +96,13 @@ def run_pcw(directory, **kwargs):
+                                       start=kwargs.pop('start', 0), stop=kwargs.pop('stop', None),
+                                       force=kwargs.pop('force', False), SI=kwargs.pop('SI', False),
+                                       estimator=estimator,
+                                       method=kwargs.pop('method', 'TI'))
```
default MBAR

orbeckst commented on scripts/mdpow-pcw:
```diff
@@ -184,29 +190,40 @@ if __name__ == "__main__":
                       help="plot dV/dlambda to FILE; use png or pdf suffix to "
                       "determine the file type. 'auto' generates a pdf file "
-                      "DIRECTORY/dVdl_MOLID_pcw.pdf and 'None' disables it [%default]",
+                      "DIRECTORY/dVdl_MOLID_pow.pdf and 'None' disables it [%default]",
```
should be the original line with pcw

orbeckst commented on scripts/mdpow-pcw:
```diff
@@ -184,29 +190,40 @@ if __name__ == "__main__":
     parser.add_option('-e', '--energies', dest="energyfile",
                       help="append solvation free energies to FILE [%default]",
                       metavar="FILE")
+    parser.add_option('--estimator', dest="estimator", default='mdpow',
```
default to alchemlyb

orbeckst commented on scripts/mdpow-pcw:
```diff
@@ -184,29 +190,40 @@ if __name__ == "__main__":
                       metavar="FILE")
+    parser.add_option('--estimator', dest="estimator", default='mdpow',
+                      help="choose the estimator to be used, alchemlyb or mdpow estimators")
+    parser.add_option('--method', dest="method", default='TI',
```
default to TI

iorga replied:
Why default to TI ? We should always have MBAR as default. But maybe I miss something.

orbeckst commented on scripts/mdpow-pcw:
```diff
@@ -184,29 +190,40 @@ if __name__ == "__main__":
     parser.add_option('--force', dest='force',
                       action="store_true",
                       help="force rereading all data [%default]")
+    parser.add_option('--SI', dest='SI',
+                      action="store_true",
+                      help="Apply statistical inefficiency analysis")
```
Is SI enabled by default? Should be yes when MBAR is the default so that we automatically get good error estimates.

orbeckst replied:
If MBAR and SI is the default then Is the sentence under stride still correct, i.e., is it true with SI+MBAR that stride>1 has little impact on precision?? If in doubt, remove the sentence so that we don't provide wrong advice.

The default for stride should probably be 1 when used with SI.

iorga replied:
Stride should be 1 in all cases. The user can increase manually this value if the analysis is too slow, being aware of consequences.

stride>1 has an important impact on precision in many cases. stride=10 was an arbitrary default value, useful with TI, which now should be replaced with the result of SI.

orbeckst reviewed (changes requested):
Please make alchemlyb/MDPOW the default for all scripts and enable SI by default for all of them. This should probably have stride=1 as default to be safe.

orbeckst commented:
@iorga comments welcome

iorga commented:
@orbeckst I agree with all your comments.

@VOD555 Please also print the SI value that is used for the analysis somewhere in the log, with a keyword that makes it easy to find using grep.

orbeckst reviewed (changes requested):
Thanks for addressing the comments. The only thing that is missing (I think) is @iorga 's request that the value of SI is logged. Could you please add this somewhere – should be logged at INFO level, I think.

VOD555 commented:
> Thanks for addressing the comments. The only thing that is missing (I think) is @iorga 's request that the value of SI is logged. Could you please add this somewhere – should be logged at INFO level, I think.

Yeah, I'm currently working one the tests whose results were changed due to the changed default setting.
Current alchemlyb SI doesn't have an `g` output, I think I will rewrite the SI function in `fep`.

orbeckst commented:
Could you deduce g from how many elements are in the DataFrame before and after subsampling? I admit that this would be a hack...

orbeckst commented:
Btw, we released alchemlyb 0.4.0, which includes some code for plotting in https://alchemlyb.readthedocs.io/en/0.4.0/visualisation.html

VOD555 commented:
@orbeckst This should be ok for final review.

orbeckst commented on mdpow/fep.py:
```diff
@@ -139,8 +139,8 @@
 from alchemlyb.parsing.gmx import _extract_dataframe
-from alchemlyb.preprocessing.subsampling import statistical_inefficiency
-
+from pymbar.timeseries import (statisticalInefficiency,
```
add pymbar to setup.py

orbeckst commented on mdpow/fep.py:
```diff
@@ -1025,14 +1025,20 @@ def collect_alchemlyb(self, SI=False, start=0, stop=None, stride=None, autosave=
-                    xvg_df = statistical_inefficiency(xvg_df, series=ts, conservative=True)
+                    # calculate statistical inefficiency of series
+                    statinef  = statisticalInefficiency(ts, fast=False)
+                    logger.info("The statistical inefficiency value is {:.4f}.".format(statinef))
```
Also log the conservative value
```python
                    logger.info("The data are subsampled every {:d} frames.".format(int(np.ceil(statinef))))
```
(not sure if I have enough parentheses...)

orbeckst commented on mdpow/tests/test_analysis_alchemlyb.py:
```diff
@@ -95,7 +94,7 @@ def test_estimator_alchemlyb(self, fep_benzene_directory, method,
         G.convert_edr()
         try:
-            G.analyze_alchemlyb(force=True, autosave=False)
+            G.analyze_alchemlyb(force=True, autosave=False, SI = False)
```
```suggestion
            G.analyze_alchemlyb(force=True, autosave=False, SI=False)
```

orbeckst reviewed (changes requested):
Looks good, minor things see comments. Thanks!

orbeckst commented on mdpow/fep.py:
```diff
@@ -1025,14 +1025,21 @@ def collect_alchemlyb(self, SI=False, start=0, stop=None, stride=None, autosave=
+                    # calculate statistical inefficiency of series
+                    statinef  = statisticalInefficiency(ts, fast=False)
+                    logger.info("The statistical inefficiency value is {:.4f}.".format(statinef))
+                    logger.info("The data are subsampled every {:d} frames.".format(int(np.ceil(statinef))))
```
fails because np not imported

orbeckst reviewed (changes requested):
Sorry, my code used np, which was not imported. Therefore, the tests failed.

orbeckst commented:
Thank you @VOD555 !

orbeckst commented:
@iorga please try out the latest master branch with these changes

iorga commented:
@orbeckst I will able to test it only on Thursday or Friday (travelling).

iorga commented:
@orbeckst @VOD555

I have analyzed `V01-n-pentane-0` with `mdpow-pow` and `mdpow-pcw`. It works well (we get solvation free energies and Pow and Pcw values), but there is no information about the statistical inefficiency in the log file (see the attached file, the analysis was run on 2021-05-01).
[mdpow.log](https://github.com/Becksteinlab/MDPOW/files/6409014/mdpow.log)

I cannot evaluate if MBAR/alchemlyb was really used for the analysis as there is no information in the log about this. I propose that at the beginning of the analysis we print in the log all the options that will be used (even the default ones). In this way we will be able to know afterwards what exact parameters were used for a given analysis. Maybe we should open a separate issue for this.

iorga commented:
@orbeckst @VOD555

In the analysis I have used the `-p auto` option (`mdpow-pow -p auto V01-n-pentane`), but no .pdf file was generated in the folder.

VOD555 commented:
@iorga @orbeckst 
I've done a test run. If the SI was performed, there were information in the log. And I found `store false` would not set the default to be `True` for SI option, so actually current mdpow cannot run with SI as default. I'm adding the default setting for SI. I think I can fix it in #144 

And for the plot function, it only works with `mdpow` estimator. As the plot function needs free energy results from each lambda window, it's impoosible to have such plot with `mbar` method which generates only one result based on all windows.

iorga commented:
Thanks @VOD555 . 

I understand for the plot. In this case, all these details should be present in the USAGE message.

Also, in the same USAGE message, I saw `--noSI                Apply statistical inefficiency analysis`, but it should be `Disable statistical inefficiency analysis`.
